### ISSUE TITLE
Some area* queries don't work on dotnet-api-docs repo

### ISSUE THREAD

**Issue by ericstj** [terrajobst/issuesof.net#25]
I noticed today that the following queries didn't work:
https://issuesof.net/?q=is%3Aopen%20is%3Aissue%20area-lead%3Aericstj%20repo%3Adotnet-api-docs
https://issuesof.net/?q=is%3Aopen%20is%3Aissue%20area-owner%3Atarekgh%20repo%3Adotnet-api-docs
https://issuesof.net/?q=is%3Aopen%20is%3Aissue%20area-pod%3Aeric-jeff%20repo%3Adotnet-api-docs

I know that these *should* work since we have issues that should be included, like:
https://issuesof.net/?q=is%3Aopen%20is%3Aissue%20area%3AMeta%20repo%3Adotnet-api-docs
for the last query.

Do we need to add an area-owners file to dotnet-api-docs?  @jeffhandley @terrajobst

**Comment by jeffhandley** [comment]
Yes, for this to work we do need to have the area-owners file in dotnet-api-docs as well; good catch.

**Comment by terrajobst** [comment]
The way the index for issuesof.net is currently modeled is that the area owners are considered repo local; that is when tagging issues with `area-lead`, `area-owner`, and `area-pod` it only considers the repo-specific area owners file.

Considering that we're using org-wide teams for areas that doesn't make a whole lot of sense anymore.

I see a couple of options for this:

1. **Decentralized stored, but mapping is global**. We can keep the repo-local `area-owners.md` files but merge them all during indexing into a cental area ownership structure that is then used to tag issues. In this model, it doesn't matter which repo says that `area-meta` is owned by Eric St John, Eric will be tagged with `area-owner` on all issues that have the `area-meta` label.

2. **Centralized stored, mapping is global**. In this model we designate a specific repo that we use store ownership in. In this model, `area-owners.md` in other repos is just ignored.

3. **Hybrid**. In this model, we have a specific repo that models central ownership, but repos can choose to replace them. The replacement would be on a per-area basis, that is all repos inherit the global mapping but individual areas can be remapped.

4. **Teams only**. In this model, we get rid of the `area-owners.md` file entirely (at least for indexing purposes) and only use org-wide teams. This is like the first option except that it's expressed via teams. All teams starting with `area-*` are considered defining are ownership. Maintainers are considered leads, everyone else is considered an owner.

I'm leaning towards (4) as this keeps avoids double bookkeeping and makes it obvious that ownership is global.

Thoughts?

**Comment by jeffhandley** [comment]
For each of those options:

1. **Decentrally stored, mapped globally**. This seems like the simplest approach all-up to me. If multiple repos specify `area-meta`, and they have different owners, then the global `area-meta` would end up containing the full set of all `area-meta` owners across all repos, but queries can be filtered by repo anyway, so that's okay.

2. **Centrally stored**, mapped globally. It's nice having the area-owners.md file embedded in the repo for discoverability and maintainability.

3. **Hybrid**. That works, and we could even have dotnet/runtime serve as the default until a different central location emerges.

4. **Teams only**. Intriguing, and I'm generally in favor, but I don't think all teams across all repos would like this model since it isn't very flexible.

I recommend we go with Option 1 for now, assuming it's relatively straightforward to implement.

**Comment by terrajobst** [comment]
Sounds good; I'll implement option (1) then. It's fairly trivial.

We have to decide what we do for cases where two repos define the same area with different lead(s) and owners.

My approach would be to just merge them as this avoids losing data.

@jeffhandley considering that the GitHub `area-*` teams have seemingly overlapping but not identical mappings: would you want me to read them and merge into the structure as well (option 4) or just entirely ignore them and only read them when they are explicitly referenced in the `area-owners.md` file? I'm leaving towards the former as this allows us to model it either way and since I'm already reading teams it makes no difference to me.

**Comment by jeffhandley** [comment]
If multiple repos specify an `area-` with different owners (teams or not), then the global area would end up containing the full set of all owners across all repos, but queries can be filtered by repo anyway, so that's okay.

Reading the teams as you suggest seems OK to me. Please don't interpret "maintainer" as "lead" though, as for most teams we've made all team members who are members of the team "maintainers" of the team so maintenance of team members isn't blocked on leads. Also, @ericstj and I put both of ourselves on all our collective teams; so we'll both show up as owners to all of our collective areas, but only 1 of us should be the lead--and that would be who is marked as the lead in the markdown file.

**Comment by terrajobst** [comment]
https://github.com/terrajobst/issuesof.net/issues/25

Quick update:

* **Area ownership is now merged across all repos**. `code-owners.md` files are now merged across all repos (but we currently only have one that has the expected format, in `dotnet/runtime`).
    - If two repos define ownership for the same area, their `Lead` and `Owners` columns are individually merged.
* **`Lead` and `Owners` may contain teams**. Any team names listed in `Lead` or `Owners` will be expanded and their members will be added.
    - The individual teams are still in the list, so you can also use `area-owner:dotnet/team-name` if you want to.
* **`Lead` contain multiple teams and users.**
    - The data structure had to support this in order to remain well-formed when multiple repos define different leads for the same area.
* **Teams alone aren't used to infer area ownership**. We could do that, for example, by indexing teams with an `area-` prefix.
    - Teams themselves have no way to express area leads as Jeff didn't want to use maintainers for that purpose, which means that folks still need to use `area-owners.md` anyway, so we might as well declare those as the source of truth.
    - Also, the team names and area labels currently don't match (for instance, we have a team called `area-system-reflection` but the corresponding area label is `area-System.Reflection`. However, we could map this by normalizing the separators.
    - This means that currently the only relationship between teams and areas is established via `area-owners.md` and the naming of the teams doesn't matter.

I have also added a new [areas page](https://issuesof.net/areas) that exposes the central area ownership structure. A few things are noteworthy:

* You can hover over entries to see where they come from.
    <img width="264" alt="image" src="https://github.com/terrajobst/issuesof.net/assets/5169960/be46f3b5-6894-4e4a-a4d0-b2d0779fc06c">

* Entries that got expanded from teams are listed after and are visually de-emphasized.
   - The goals here is emphasize the users and teams that should be used for tagging while also showing the effective containment for troubleshooting.
   ![image](https://github.com/terrajobst/issuesof.net/assets/5169960/46166439-d1ea-48e4-a5fb-83b1a248defd)
